src/Compilers/CSharp/Portable/Binder/SubsumptionDiagnosticBuilder.cs

```diff
@@ -272,9 +272,13 @@ private ErrorCode CheckSubsumed(BoundPattern pattern, DecisionTree decisionTree,
                         }
                     }
                 case BoundKind.DiscardPattern:
-                    // because we always handle `default:` last, and that is the only way to get a wildcard pattern,
-                    // we should never need to see if it subsumes something else.
-                    throw ExceptionUtilities.UnexpectedValue(decisionTree.Kind);
+                    if (pattern.Syntax.Kind() == SyntaxKind.DefaultSwitchLabel)
```

Comment:
No, there will not be a `default` case in the pattern switch. However, you can simply use `_` instead. The difference is that the `default` label is always last. The wildcard `_` is taken in the order it is placed, and not assumed to be reachable (we check).

Also, you won't be permitted to use `default` as a constant in a constant pattern, to reduce confusion that it might be perceived to be a default/match-all pattern.

---
In reply to: [166480340](https://github.com/dotnet/roslyn/pull/23357#discussion_r166480340) [](ancestors = 166480340)